lib/nrf_modem_lib/nrf_modem_lib_trace.c

```diff
@@ -89,35 +192,22 @@ int nrf_modem_lib_trace_start(enum nrf_modem_lib_trace_mode trace_mode)
 int nrf_modem_lib_trace_process(const uint8_t *data, uint32_t len)
 {
 	if (!is_transport_initialized) {
```

Comment:
I looked in this. The `is_transport_initialized` can be false if the transport initialization had failed (for example, when uart_init() failed). In that case, we did attempt an initialization but that failed. In that case, I prefer that this API returns failure (and free up trace memory) instead of asserting. The reason is that transport initialization can fail because of external factors. I prefer that nrf91 application does not assert/crash because of that. A LOG would be sufficient. What do you think?